core/controller/project.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * core/controller/project.js
+ * Simple project handling (until official API is set)
+ *
+ * @exports {Object} default - Koa router
+ */
+
+import Router from 'koa-router'
+
+import { Project } from '~/core/model/project'
+import { Release } from '~/core/model/release'
+import { Cycle } from '~/core/model/cycle'
+import { Build } from '~/core/model/build'
+import { IsRole } from '~/core/policy/isRole'
+import { GetReleases } from '~/core/service/github'
+
+// TODO: GitHub hook integration
+let route = new Router({
+  prefix: '/project/:owner/:name'
+})
+
+// TODO: we should probably do some data checks here. for 'security' and stuff
+route.get('/*', IsRole('BETA'), async (ctx, next) => {
+  ctx.project = await Project.findOne({
+    'github.owner': ctx.params.owner,
+    'github.name': ctx.params.name
+  })
+
+  if (ctx.project == null) {
+    return ctx.throw('The project you are looking for has been lost in space', 404)
+  }
+
+  await next()
+})
+
+route.get('/init', async (ctx, next) => {
+  const status = await ctx.project.getStatus()
+  .catch(err => ctx.throw({
+    message: 'Unable to get project status',
+    error: err
+  }, 500))
+
+  if (status !== 'NEW') return ctx.throw('The project is already in standby', 400)
```

Comment:
Well if the project's status isn't 'NEW' then we can assume we already setup all of the hooks (eventually), GitHub label, and stuff we need to. As such, all I can think of to say is 'The project is already initialized'